hermes-common/src/main/java/pl/allegro/tech/hermes/infrastructure/zookeeper/ZookeeperSubscriptionOffsetChangeIndicator.java

```diff
@@ -59,6 +60,33 @@ public PartitionOffsets getSubscriptionOffsets(TopicName topic, String subscript
         return allOffsets;
     }
 
+    @Override
+    public boolean areOffsetsMoved(TopicName topicName, String subscriptionName, String brokersClusterName,
+                                   KafkaTopic kafkaTopic, List<Integer> partitionIds) {
+        return partitionIds.stream().allMatch(partitionId -> offsetDoesNotExist(topicName, subscriptionName, brokersClusterName, partitionId, kafkaTopic));
+    }
+
+    @Override
+    public void removeOffset(TopicName topicName, String subscriptionName, String brokersClusterName, KafkaTopicName kafkaTopicName, int partitionId) {
+        String offsetPath = paths.offsetPath(topicName, subscriptionName, kafkaTopicName, brokersClusterName, partitionId);
+
+        try {
+            zookeeper.delete().guaranteed().deletingChildrenIfNeeded().forPath(offsetPath);
```

Comment:
Should we retry this operation?

Reply:
User will receive an error if retransmission fails. After the fail user can repeat the operation by itself, therefore we dont have to add additional retry here. At least now.